### PR TITLE
 Unify ocf_env for regular use and unit tests

### PR DESCRIPTION
**Pull request by Ostrokrzew**
**ocf_env**: unified and deleted separate files for regular use and UT,
now UT uses the same files as 'regular' OCF;
changed 'Intel OCF' on 'OCF' due to earlier suggestion.
**prepare script**: 'CMakeLisits.txt' now would add necessary
for UT libraries during compilation;
changed omitted phrase 'python3.6' on 'python3';
removing 'static inline' keywords from UT's copy of ocf_env.h file,
due to compilation issues.
**run_unit_tests**: symlinking ocf_env files to test directory at start;
added cleanup function on startup, before actual start of UT.
**tests_config**: added path to ocf_env directory.
**alru cleaning policy**: little update in functions to properly use
atomic values after unify of ocf_env.
**general**: fixed few typos;
deleted recursive inclusions;
some aesthetic changes in comments.
**add new test**: added missing parentheses in one of 'print' functions;

Signed-off-by: Slawomir_Jankowski <slawomir.jankowski@intel.com>

### CODE REVIEW

File: env/posix/ocf_env.h [hunk 1]
```diff
@@ -49,31 +55,72 @@ typedef uint64_t u64;
...
+#define env_strnlen(s, smax) strnlen(s, smax)
+#define env_strncmp strncmp
+#define env_strncpy(dest, dmax, src, slen) ({ \
+		strncpy(dest, src, min(dmax - 1, slen)); \
+		dest[dmax - 1] = '\0'; \
+		0; \
+	})
+
+#ifdef UNIT_TEST
+int env_memcpy(void *dest, size_t destsz, const void * src, size_t count)
```
Comment (arutk) [inline env/posix/ocf_env.h]: isn't this functionally identical to the other version (line 84) ?

File: env/posix/ocf_env.h [hunk 2]
```diff
@@ -49,31 +55,72 @@ typedef uint64_t u64;
...
+{
+	if (destsz < count)
+		memcpy(dest, src, destsz);
+	else
+		memcpy(dest, src, count);
+	return 0;
+}
+
+#ifdef UNSTATIC
```
Comment (arutk) [inline env/posix/ocf_env.h]: this is repeated in line 644

Comment (Ostrokrzew) [reply env/posix/ocf_env.h]: in line 644 this is undefined

File: env/posix/ocf_env.h (hunk 1 above)
Comment (Ostrokrzew) [reply env/posix/ocf_env.h]: Indeed, but UT can't handle this function as a macro

Comment (Ostrokrzew) [reply env/posix/ocf_env.h]: fixed

File: env/posix/ocf_env.c [hunk 3]
```diff
@@ -6,7 +6,11 @@
 #include "ocf_env.h"
 #include <sched.h>
 #include <execinfo.h>
+#include <sys/types.h>
+#include <setjmp.h>
+#include <cmocka.h>
```
Comment (arutk) [inline env/posix/ocf_env.c]: We shouldn't include cmocka headers in posix env files. Not sure about the setjmp.h - if this is specific to test framework, it shouldn't be here either.

File: env/posix/ocf_env.h [hunk 4]
```diff
@@ -647,10 +623,13 @@ struct env_timeval {
 
 uint32_t env_crc32(uint32_t crc, uint8_t const *data, size_t len);
 
-#define ENV_PRIu64 "lu"
-
 unsigned env_get_execution_context(void);
 void env_put_execution_context(unsigned ctx);
 unsigned env_get_execution_context_count(void);
 
+#ifdef UNSTATIC
```
Comment (arutk) [inline env/posix/ocf_env.h]: is it really necessary to have #ifdef UNSTATIC in this file?

File: tests/unit/framework/prepare_sources_for_testing.py [hunk 5]
```diff
@@ -158,7 +160,11 @@ def prepare_autowraps(self, test_file_path, preprocessed_file_path):
         with open(preprocessed_file_path) as f:
             code = f.readlines()
             for function in functions_to_wrap:
-                if function.startswith("env_") or function.startswith("bug"):
+                if function.startswith("env_") or function.startswith("bug") \
+                   or function.startswith("memcpy"):
+                    # added memcpy function to list of ignored functions,
+                    # because earlier it was regular function
```
Comment (arutk) [inline tests/unit/framework/prepare_sources_for_testing.py]: I think you don't need to describe the previous implementation here - it is enough to state that this is a macro

File: env/posix/ocf_env.h (hunk 4 above)
Comment (Ostrokrzew) [reply env/posix/ocf_env.h]: removed

File: tests/unit/framework/prepare_sources_for_testing.py (hunk 5 above)
Comment (Ostrokrzew) [reply tests/unit/framework/prepare_sources_for_testing.py]: fixed

File: env/posix/ocf_env.c (hunk 3 above)
Comment (Ostrokrzew) [reply env/posix/ocf_env.c]: Removed and added during UT compilation.

**Comment by Ostrokrzew** [comment]
retest this please

**Comment by Ostrokrzew** [comment]
retest this please

**Comment by Ostrokrzew** [comment]
retest this please

File: tests/unit/tests/ocf_env/CMakeLists.txt [hunk 6]
```diff
@@ -1,2 +1,3 @@
-add_library(ocf_env ocf_env.c)
-target_link_libraries(ocf_env pthread)
+add_library(ocf_env ocf_env.c /usr/include/sys/types.h /usr/include/setjmp.h /usr/include/cmocka.h)
+add_definitions(-Dstatic=  -Dinline=  )
+target_link_libraries(ocf_env pthread z)
```
Comment (arutk) [inline tests/unit/tests/ocf_env/CMakeLists.txt]: is this 'z' intentional?

**Comment by Ostrokrzew** [comment]
retest this please

**Comment by Ostrokrzew** [comment]
retest this please

**Comment by KamilLepek** [comment]
@CAS-Linux-Jenkins retest this please

**Comment by KamilLepek** [comment]
@CAS-Linux-Jenkins retest this please

**Comment by micrakow** [comment]
@CAS-Linux-Jenkins retest this please

**Pull request merged**